server/src/main/scala/org/scassandra/server/actors/TcpServer.scala

```diff
@@ -26,20 +26,30 @@ import org.scassandra.server.priming.ActivityLog
 import org.scassandra.server.priming.batch.PrimeBatchStore
 import org.scassandra.server.priming.prepared.PreparedStoreLookup
 import org.scassandra.server.priming.query.PrimeQueryStore
-import org.scassandra.server.{RegisterHandler, ServerReady}
+import org.scassandra.server.{ShutdownServer, Shutdown, RegisterHandler, ServerReady}
+import spray.can.Http
 
 import scala.concurrent.duration._
 import scala.language.postfixOps
 
 import scala.concurrent.Future
 
-class TcpServer(manager: ActorRef,
-                listenAddress: String, port: Int,
+class TcpServer(listenAddress: String, port: Int,
                 primedResults: PrimeQueryStore,
                 primePrepareStore: PreparedStoreLookup,
                 primeBatchStore: PrimeBatchStore,
                 serverReadyListener: ActorRef,
-                activityLog: ActivityLog) extends Actor with ActorLogging {
+                activityLog: ActivityLog,
+                manager : Option[ActorRef]) extends Actor with ActorLogging {
+
+  def this(listenAddress: String, port: Int,
```

Comment:
Seems weird to have 2 constructors here when i could have given a default value to `manager`, but unfortunately Akka does not seem to handle when Properties are provided without a value for a argument that has a default value.